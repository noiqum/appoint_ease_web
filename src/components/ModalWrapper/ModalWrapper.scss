.modal-wrapper {
  position: relative;
  z-index: 1000;
  &.close {
    display: block;
  }
  &.active {
    position: fixed;
    top: 0;
    left: 0;
    width: 100vw;
    height: 100vh;
    background-color: rgba(0, 0, 0, 0.301);
  }
}
