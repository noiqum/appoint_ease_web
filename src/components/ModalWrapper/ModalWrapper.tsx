import { useAppDispatch, useAppSelector } from '../../store/hooks'
import './ModalWrapper.scss'
import { RootState } from '../../store/index'
import React from 'react'
import { setOpenStatus } from '../../store/modalSlice'

interface IProps {
  children?: React.ReactNode
}
export const ModalWrapper: React.FC<IProps> = ({ children }: IProps) => {
  const isOpen = useAppSelector((state: RootState) => state.modal.isOpen)
  const modalElement = useAppSelector((state: RootState) => state.modal.modalElement)
  const dispatch = useAppDispatch()
  return (
    <div
      onClick={() => isOpen && dispatch(setOpenStatus(false))}
      className={isOpen ? 'modal-wrapper active' : 'modal-wrapper close'}
    >
      {isOpen && modalElement}
      {children}
    </div>
  )
}
