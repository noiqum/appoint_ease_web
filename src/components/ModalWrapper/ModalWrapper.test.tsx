import { ModalWrapper } from './ModalWrapper'
import { renderWithProviders } from '../../test-utils'

describe('ModalWrapper', () => {
  it('renders', () => {
    const { asFragment } = renderWithProviders(<ModalWrapper />)
    expect(asFragment()).toMatchSnapshot()
  })
})
