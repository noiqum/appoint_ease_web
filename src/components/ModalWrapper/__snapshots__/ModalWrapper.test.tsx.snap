// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWrapper renders 1`] = `
<DocumentFragment>
  <div
    class="modal-wrapper close"
  />
</DocumentFragment>
`;
